File: mobile/lib/dtos/register_request.dart
class RegisterRequest {
  String username;
  String email;
  String password;
  String passwordConfirmation;

  RegisterRequest({
    required this.username,
    required this.email,
    required this.password,
    required this.passwordConfirmation,
  });

  factory RegisterRequest.fromJson(Map<String, dynamic> json) =>
      RegisterRequest(
        username: json["username"],
        email: json["email"],
        password: json["password"],
        passwordConfirmation: json["passwordConfirmation"],
      );

  Map<String, dynamic> toJson() => {
        "username": username,
        "email": email,
        "password": password,
        "passwordConfirmation": passwordConfirmation,
      };
}

File: backend/MemoWise.Model/DTO/RegisterRequest.cs
using System.ComponentModel.DataAnnotations;

namespace MemoWise.Model.DTO;

public class RegisterRequest
{
    [Required]
    [MaxLength(50)]
    public string Username { get; set; }
    [Required]
    public string Email { get; set; }
    [Required]
    [MinLength(6)]
    public string Password { get; set; }
    [Required]
    public string PasswordConfirmation { get; set; }
}
